packages/server/lib/cloud/api/axios_middleware/encryption.ts

```diff
@@ -0,0 +1,110 @@
+import type { AxiosError, AxiosInstance, AxiosResponse, InternalAxiosRequestConfig } from 'axios'
+import * as enc from '../../encryption'
+import { PUBLIC_KEY_VERSION } from '../../constants'
+import crypto, { KeyObject } from 'crypto'
+import { DecryptionError } from '../cloud_request_errors'
+import axios from 'axios'
+
+let encryptionKey: KeyObject
+
+declare module 'axios' {
+  interface AxiosRequestConfig {
+    encrypt?: 'always' | 'signed' | boolean
+  }
+}
+
+const maybeEncryptRequest = async (req: InternalAxiosRequestConfig) => {
```

Comment:
Why is this "maybe"?

Reply:
Ah, because previously I wasn't using the axios's `runWhen` and had the conditional inlined, so it was always hitting this but only sometimes encrypting. Can reword